contrib/azure/azure-coreos-cluster

```diff
@@ -220,11 +220,14 @@ for i in range(args.num_nodes):
 
     try:
       if i == 0:
-          result = sms.create_virtual_machine_deployment(
+          try:
+            result = sms.create_virtual_machine_deployment(
                       args.cloud_service_name, deployment_name=args.cloud_service_name,
                       deployment_slot='production', label=vm_name,
-                      role_name=vm_name, system_config=system, os_virtual_hard_disk=os_hd,
+                      role_name=vm_name, system_config=system, os_virtual_hard_disk=os_hd, virtual_network_name=args.virtual_network_name,
```

Comment:
Only got the error when subnet-name was set. Had to review the python client code to find this parameter, which interestingly sms.add_role doesn't even have the option. It appears to attach to the same virtual network that the virtual machine deployment was attached to.